### ISSUE TITLE
Remove the unreachable F0.ndim == 0 branch in BaseBoosting.predict

### ISSUE THREAD

**Issue by sachncs** [sachncs/gradient-regularized-newton-boosting-trees#17]
## Problem

`BaseBoosting.predict` has a dead branch for `self.F0.ndim == 0` that can never be reached, because `init_prediction` always returns a 1-D array of zeros (matching the shape of `y`). The 0-D branch's logic (extract scalar via `.item()`) is therefore untested, and the comment about it ("When F0 was stored as a 0-D scalar (the init_prediction default)...") describes behavior the codebase never exhibits.

## Evidence

- `grnbt/boosting.py:243-251` — the `if self.F0.ndim == 0:` branch.
- `grnbt/boosting.py:272-273` — `init_prediction` returns `np.zeros_like(y, dtype=float)`, which has `y.ndim == 1` since `validate_fit_inputs` requires `y.ndim == 1`.
- `grnbt/boosting.py:308-310` — `validate_fit_inputs` enforces `y.ndim == 1`, so `init_prediction(y)` always returns a 1-D array.

Conclusion: `self.F0.ndim == 0` is unreachable in the current code path.

## Root Cause

The 0-D branch was added speculatively ("What if F0 is a scalar?") without verifying the actual code path. `init_prediction` always returns a 1-D array, and `predict` only consumes the *mean* of F0 (not its per-sample values), so the only meaningful state is the 1-D case.

## Impact

- Dead code: a branch that cannot be reached under the current API.
- Hidden semantic mismatch: `init_prediction` returns a vector but `predict` only uses the mean, so the documented "warm-start at mean of y" is silently implemented as "broadcast mean to all samples".
- Source of confusion for maintainers reading the predict() code.

## Recommendation

Two clean options:

**Option A (preferred) — drop the 0-D branch**:

```python
def predict(self, x):
    ...
    baseline = float(np.mean(self.F0))
    f_out = np.full(x.shape[0], baseline, dtype=float)
    for tree in self.trees:
        f_out += self.learning_rate * tree.predict(x)
    return f_out
```

**Option B — actually support a scalar F0**:

Change `init_prediction` to optionally return a scalar (e.g., for MSE warm-start where the optimal F₀ is the mean of y), and document the convention. Then the 0-D branch becomes reachable.

If picking Option A, also add a docstring to `init_prediction` stating that the returned array's *mean* is what's used at predict-time, not its per-sample values, so a subclass overriding to a different baseline must account for that.

## Acceptance Criteria

- [ ] The unreachable branch is removed (Option A) or made reachable (Option B).
- [ ] A test asserts the chosen behavior (e.g., "predict on new data uses mean(F0) + sum(trees)").
- [ ] `init_prediction` docstring documents the actual contract.

## Verification

Run `pytest tests/test_boosting.py -v` and confirm no regression.

## Dependencies

None.

## Risk

Low. The behavior is unchanged for current usage; this is a cleanup.

**Comment by sachncs** [comment]
Resolved in PR #40.
